engine/src/main/java/com/redhat/cloud/notifications/exports/ExportsService.java

```diff
@@ -0,0 +1,91 @@
+     * @param resourceUuid the {@link UUID} of the requested resource.
+     * @param exportContents the payload of the request.
+     */
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/app/export/v1/{exportRequestUuid}/{application}/{resourceUuid}/upload")
+    @POST
+    @Retry(maxRetries = 3)
+    void uploadJSONExport(
+        @HeaderParam(Constants.X_RH_EXPORT_SERVICE_PSK) String xRhExportServicePsk,
+        @RestPath UUID exportRequestUuid,
+        @RestPath String application,
+        @RestPath UUID resourceUuid,
+        String exportContents
+    );
+
+    /**
+     * Sends a CSV payload to the export service.
+     * @param xRhExportServicePsk the export service PSK required for the authorization.
+     * @param exportRequestUuid the {@link UUID} of the export request.
+     * @param application the application the export request got requested to.
+     * @param resourceUuid the {@link UUID} of the requested resource.
+     * @param exportContents the payload of the request.
+     */
+    @Consumes("text/csv")
+    @Path("/app/export/v1/{exportRequestUuid}/{application}/{resourceUuid}/upload")
+    @POST
+    @Retry(maxRetries = 3)
+    void uploadCSVExport(
+        @HeaderParam(Constants.X_RH_EXPORT_SERVICE_PSK) String xRhExportServicePsk,
+        @RestPath UUID exportRequestUuid,
+        @RestPath String application,
+        @RestPath UUID resourceUuid,
+        String exportContents
+    );
+
+    /**
+     * Sends an error signal to the export service about a failed export
+     * request.
+     * @param xRhExportServicePsk the export service PSK required for the authorization.
+     * @param exportRequestUuid the {@link UUID} of the export request.
+     * @param application the application the export request got requested to.
+     * @param resourceUuid the {@link UUID} of the requested resource.
+     * @param exportError the error payload.
+     */
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/app/export/v1/{exportRequestUuid}/{application}/{resourceUuid}/error")
+    @POST
+    @Retry(maxRetries = 3)
+    void notifyErrorExport(
+        @HeaderParam(Constants.X_RH_EXPORT_SERVICE_PSK) String xRhExportServicePsk,
+        @RestPath UUID exportRequestUuid,
+        @RestPath String application,
+        @RestPath UUID resourceUuid,
+        ExportError exportError
+    );
+
+    /**
+     * Throws a runtime exception with the client's response for an easier
+     * debugging.
+     * @param response the received response from the Export Service.
+     * @return the {@link RuntimeException} to be thrown.
+     *
+    @ClientExceptionMapper
+    static RuntimeException toException(final Response response) {
+        final String errMessage = String.format("The export service responded with a %s status: %s", response.getStatus(), response.readEntity(String.class));
+
+        throw new WebApplicationException(errMessage, response);
+    }*/
```

Comment:
Should this be uncommented before the merge?

Reply:
Yes! There's a couple of things I need to finish in this PR:

1. This.
2. Update the schema.